src/chrome/extension/scripts/chrome_browser_api.ts

```diff
@@ -201,10 +201,16 @@ class ChromeBrowserApi implements BrowserAPI {
     this.popupState_ = PopupState.LAUNCHED;
   }
 
-  public showNotification = (notificationText :string) => {
+  public showNotification = (text :string, tag :string) => {
     var notification =
-        new Notification('uProxy', { body: notificationText,
-                         icon: 'icons/38_' + UI.DEFAULT_ICON});
+        new Notification('uProxy', {
+          body: text,
+          icon: 'icons/38_' + UI.DEFAULT_ICON,
+          tag: tag
+        });
+    notification.onclick = function() {
+      ui.handleNotificationClick(this.tag);
```

Comment:
ui is moving to a popup, while browser api will stay in background page, so you won't have access to ui here.

There's really no point in checking this in.